src/functions.php

```diff
@@ -2,9 +2,77 @@
 
 namespace Amp\Http\Server;
 
+use Amp\ByteStream\InputStream;
 use Amp\Http\Status;
+use Amp\Loop;
+use Amp\Socket\Server as Socket;
+use Amp\Socket\SocketException;
+use Generator;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
 
 function redirectTo(string $uri, int $statusCode = Status::FOUND): Response
 {
     return new Response($statusCode, ['location' => $uri]);
 }
+
+/**
+ * Returns a response with the html content type.
+ *
+ * @param InputStream|string|null $content
+ * @param int $status
+ * @return Response
+ */
+function html($content, int $status = Status::OK): Response
+{
+    return new Response($status, [
+        'Content-Type' => 'text/html; charset=utf-8'
+    ], $content);
+}
+
+/**
+ * Creates an HTTP server.
+ *
+ * This function is inspired in Golang's http module one. It provides sensible defaults for a really quick start
+ * and it is designed to cover 90% of use cases.
+ *
+ * If you need fine-tuning, like listening in two sockets or attach more listeners to the server
+ * you should bootstrap your server manually.
+ *
+ * @param string $address
+ * @param RequestHandler $handler
+ * @param Options|null $options
+ * @param LoggerInterface|null $logger
+ * @return Generator
+ * @throws Loop\UnsupportedFeatureException
+ * @throws SocketException
+ */
+function listenAndServe(string $address, RequestHandler $handler, Options $options = null, LoggerInterface $logger = null): Generator
```

Comment:
Great, I'll fix it.

Reply:
You could allow `$address` to be an array and listen to multiple sockets that way. Thoughts @kelunik ?

Reply:
Also for options and logger I'd add question marks to the typehints.

Personally I'd also drop the `@param` annotations since they don't add anything and only duplicate the typehints anyway. But then again current Amphp code usually has those duplications so maybe @trowski and @kelunik want them?

Reply:
We usually only have them if there's also a description or more specific types.

Reply:
> You could allow `$address` to be an array and listen to multiple sockets that way. Thoughts @kelunik ?

Most of the servers I've implemented just listen in one socket. But I'm not against the idea. I think the function will work for the vast majority of use cases.

Reply:
> Most of the servers I've implemented just listen in one socket.

These days I'd argue to listen on both IPv4 and IPv6.

Reply:
@kelunik Usually eh? Idk about that... [this](https://github.com/enumag/http-server/commit/fdddbfbf17a0c02bb3bacbef0189dbeb1cb5ceb6) is just one file... Can I go and wipe them out?